src/aws-cpp-sdk-core/source/endpoint/DefaultEndpointProvider.cpp

```diff
@@ -138,6 +138,21 @@ ResolveEndpointDefaultImpl(const Aws::Crt::Endpoints::RuleEngine& ruleEngine,
                 AWS_LOGSTREAM_TRACE(DEFAULT_ENDPOINT_PROVIDER_TAG, "Endpoint str eval parameter: " << parameter.GetName() << " = " << parameter.GetStrValueNoCheck());
                 crtRequestCtx.AddString(Aws::Crt::ByteCursorFromCString(parameter.GetName().c_str()), Aws::Crt::ByteCursorFromCString(parameter.GetStrValueNoCheck().c_str()));
             }
+            else if(EndpointParameter::ParameterType::STRING_ARRAY == parameter.GetStoredType())
+            {
+                
+                Aws::Crt::Vector<Aws::Crt::ByteCursor> byteCursorArray;
+                byteCursorArray.reserve(parameter.GetStrArrayValueNoCheck().size());
+                Aws::OStringStream os;
+                for (const auto &e: parameter.GetStrArrayValueNoCheck())
+                {
+                    os<<e<<",";
+                    byteCursorArray.emplace_back(Aws::Crt::ByteCursorFromCString(e.c_str()));
+                }
+
+                AWS_LOGSTREAM_TRACE(DEFAULT_ENDPOINT_PROVIDER_TAG, "Endpoint str array eval parameter: " << parameter.GetName() << " = " << os.str());
```

Comment:
should move os stream construction to a [IIFE](https://en.wikipedia.org/wiki/Immediately_invoked_function_expression) so that the OStream is only constructed and used when the [log level is trace](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/logging/LogMacros.h#L74)
```cpp
AWS_LOGSTREAM_TRACE(DEFAULT_ENDPOINT_PROVIDER_TAG, 
  "Endpoint str array eval parameter: " << 
    parameter.GetName() << " = " << 
    []() ->  Aws::OStringStream {
      Aws::OStringStream os;
      for (const auto &e: parameter.GetStrArrayValueNoCheck())
     {
        os<<e<<",";
     }
     return os;
     }());
```

Reply:
ack.